### PR TITLE
Feature/impl 1341

### PR DESCRIPTION
**Pull request by fedecl-vv**
Build Automation: Submit appeal with valid denial code from Public link

### CODE REVIEW

File: Keywords/pages/Page_Licensure_Unit_Appeals.groovy [hunk 1]
```diff
@@ -6,6 +6,8 @@ import static com.kms.katalon.core.testdata.TestDataFactory.findTestData
 import static com.kms.katalon.core.testobject.ObjectRepository.findTestObject
 import static com.kms.katalon.core.testobject.ObjectRepository.findWindowsObject
 
+import javax.swing.KeyStroke
```
Comment (patriciorodriguezOT) [inline Keywords/pages/Page_Licensure_Unit_Appeals.groovy]: Let's remove the line if it is not used

Comment (patriciorodriguezOT) [inline Keywords/pages/Page_Licensure_Unit_Appeals.groovy]: Let's remove the line if it is not used

File: Keywords/pages/Page_Licensure_Unit_Appeals.groovy [hunk 2]
```diff
@@ -20,38 +22,120 @@ import com.kms.katalon.core.webui.keyword.WebUiBuiltInKeywords as WebUI
 import com.kms.katalon.core.webui.keyword.builtin.WaitForElementVisibleKeyword
 import com.kms.katalon.core.windows.keyword.WindowsBuiltinKeywords as Windows
 import com.kms.katalon.core.util.KeywordUtil
+import org.openqa.selenium.Keys as Keys
```
Comment (patriciorodriguezOT) [inline Keywords/pages/Page_Licensure_Unit_Appeals.groovy]: Let's remove the line if it is not used

Comment (patriciorodriguezOT) [inline Keywords/pages/Page_Licensure_Unit_Appeals.groovy]: Let's remove the line if it is not used

File: Keywords/pages/Page_Licensure_Unit_Appeals.groovy [hunk 3]
```diff
@@ -20,38 +22,120 @@ import com.kms.katalon.core.webui.keyword.WebUiBuiltInKeywords as WebUI
...
+	 * Buttons
+	 *************/
+	@Keyword
+	def clickOnNextBtn () {
+		WebUI.waitForElementVisible(btnNext, 10)
+		WebUI.click(btnNext)
+	}
+
+	@Keyword
+	def clicOnCloseModalBtn () {
```
Comment (patriciorodriguezOT) [inline Keywords/pages/Page_Licensure_Unit_Appeals.groovy]: @fedecl-ot We already have keywords to handle Modal buttons. Please check on Keywords>page>ModalDialog
Let's use them.

Comment (patriciorodriguezOT) [inline Keywords/pages/Page_Licensure_Unit_Appeals.groovy]: @fedecl-ot We already have keywords to handle Modal buttons. Please check on Keywords>page>ModalDialog
Let's use them.

File: Keywords/pages/Page_Licensure_Unit_Appeals.groovy [hunk 4]
```diff
@@ -20,38 +22,120 @@ import com.kms.katalon.core.webui.keyword.WebUiBuiltInKeywords as WebUI
...
+		WebUI.waitForElementVisible(btnModalClose, 10)
+		WebUI.click(btnModalClose)
+	}
+
+	@Keyword
+	def clicOnModalSubmitBtn () {
```
Comment (patriciorodriguezOT) [inline Keywords/pages/Page_Licensure_Unit_Appeals.groovy]: We already have keywords to handle Modal buttons. Please check on Keywords>page>ModalDialog
Let's use them.

Comment (patriciorodriguezOT) [inline Keywords/pages/Page_Licensure_Unit_Appeals.groovy]: We already have keywords to handle Modal buttons. Please check on Keywords>page>ModalDialog
Let's use them.

File: Keywords/pages/Page_Licensure_Unit_Appeals.groovy [hunk 5]
```diff
@@ -20,38 +22,120 @@ import com.kms.katalon.core.webui.keyword.WebUiBuiltInKeywords as WebUI
...
+		WebUI.waitForElementVisible(btnModalSubmit, 10)
+		WebUI.click(btnModalSubmit)
+	}
+	
+	@Keyword
+	def clicOnSubmitBtn () {
```
Comment (patriciorodriguezOT) [inline Keywords/pages/Page_Licensure_Unit_Appeals.groovy]: Please, update the keyword's name. It should be "click" not "clic"

Comment (patriciorodriguezOT) [inline Keywords/pages/Page_Licensure_Unit_Appeals.groovy]: Please, update the keyword's name. It should be "click" not "clic"

File: Keywords/pages/Page_Licensure_Unit_Appeals.groovy [hunk 6]
```diff
@@ -20,38 +22,120 @@ import com.kms.katalon.core.webui.keyword.WebUiBuiltInKeywords as WebUI
...
+	def verifyAppealSubmitted () {
+		WebUI.waitForElementVisible(h1AppealSubmitted, 10)
+		if (!WebUI.verifyElementPresent(h1AppealSubmitted, 10)) {
+			KeywordUtil.markFailed("\nThe test case can not be completed. Error alert won't show due INVALID denial code entered")
+		}
 	}
 
 	@Keyword
 	def verifyErrorAlert (String alertObj, String errorText) {
+		WebUI.waitForElementVisible(alertObj, 10)
```
Comment (patriciorodriguezOT) [inline Keywords/pages/Page_Licensure_Unit_Appeals.groovy]: @fedecl-ot , as we saw, you are waiting for a String. alertObj is not an Object Repository

Comment (patriciorodriguezOT) [inline Keywords/pages/Page_Licensure_Unit_Appeals.groovy]: @fedecl-ot , as we saw, you are waiting for a String. alertObj is not an Object Repository

File: Scripts/Appeal/Submit appeal with valid denial code from Public link/Script1648236418447.groovy [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+import com.kms.katalon.core.model.FailureHandling as FailureHandling
+import com.kms.katalon.core.testcase.TestCase as TestCase
+import com.kms.katalon.core.testdata.TestData as TestData
+import com.kms.katalon.core.testng.keyword.TestNGBuiltinKeywords as TestNGKW
+import com.kms.katalon.core.testobject.TestObject as TestObject
+import com.kms.katalon.core.webservice.keyword.WSBuiltInKeywords as WS
+import com.kms.katalon.core.webui.keyword.WebUiBuiltInKeywords as WebUI
+import com.kms.katalon.core.windows.keyword.WindowsBuiltinKeywords as Windows
+import internal.GlobalVariable as GlobalVariable
+import org.openqa.selenium.Keys as Keys
```
Comment (patriciorodriguezOT) [inline Scripts/Appeal/Submit appeal with valid denial code from Public link/Script1648236418447.groovy]: Let's remove this line if we are not using it.

Comment (patriciorodriguezOT) [inline Scripts/Appeal/Submit appeal with valid denial code from Public link/Script1648236418447.groovy]: Let's remove this line if we are not using it.

**Review by patriciorodriguezOT — CHANGES_REQUESTED** [review]

File: Keywords/pages/Page_Licensure_Unit_Appeals.groovy (hunk 1 above)
Comment (fedecl-ot) [reply Keywords/pages/Page_Licensure_Unit_Appeals.groovy]: Done

Comment (fedecl-ot) [reply Keywords/pages/Page_Licensure_Unit_Appeals.groovy]: Done

File: Keywords/pages/Page_Licensure_Unit_Appeals.groovy (hunk 2 above)
Comment (fedecl-ot) [reply Keywords/pages/Page_Licensure_Unit_Appeals.groovy]: Done

Comment (fedecl-ot) [reply Keywords/pages/Page_Licensure_Unit_Appeals.groovy]: Done

File: Keywords/pages/Page_Licensure_Unit_Appeals.groovy (hunk 3 above)
Comment (fedecl-ot) [reply Keywords/pages/Page_Licensure_Unit_Appeals.groovy]: Done

Comment (fedecl-ot) [reply Keywords/pages/Page_Licensure_Unit_Appeals.groovy]: Done

File: Keywords/pages/Page_Licensure_Unit_Appeals.groovy (hunk 4 above)
Comment (fedecl-ot) [reply Keywords/pages/Page_Licensure_Unit_Appeals.groovy]: Done

Comment (fedecl-ot) [reply Keywords/pages/Page_Licensure_Unit_Appeals.groovy]: Done

File: Keywords/pages/Page_Licensure_Unit_Appeals.groovy (hunk 5 above)
Comment (fedecl-ot) [reply Keywords/pages/Page_Licensure_Unit_Appeals.groovy]: Done

File: Keywords/pages/Page_Licensure_Unit_Appeals.groovy (hunk 6 above)
Comment (fedecl-ot) [reply Keywords/pages/Page_Licensure_Unit_Appeals.groovy]: Done

Comment (fedecl-ot) [reply Keywords/pages/Page_Licensure_Unit_Appeals.groovy]: Done

File: Scripts/Appeal/Submit appeal with valid denial code from Public link/Script1648236418447.groovy (hunk 7 above)
Comment (fedecl-ot) [reply Scripts/Appeal/Submit appeal with valid denial code from Public link/Script1648236418447.groovy]: Done

Comment (fedecl-ot) [reply Scripts/Appeal/Submit appeal with valid denial code from Public link/Script1648236418447.groovy]: Done

**Review by patriciorodriguezOT — APPROVED** [review]
The changes were applied correctly.

**Pull request merged**